arch/arm/core/aarch32/cortex_m/mpu/arm_mpu_v7_internal.h

```diff
@@ -216,7 +217,11 @@ static inline int is_in_region(u32_t r_index, u32_t start, u32_t size)
 			MPU_RASR_SIZE_Pos) + 1;
 	r_addr_end = r_addr_start + (1UL << r_size_lshift) - 1;
 
-	if (start >= r_addr_start && (start + size - 1) <= r_addr_end) {
+	if (size == 0 || __builtin_add_overflow(start, size - 1, &end)) {
```

Comment:
For ease of reading, can you also add brackets around the operations in all the "if" blocks, to explicitly indicate precedence?

Reply:
can do that